Documentation/guides/profiling.md

```diff
@@ -0,0 +1,244 @@
+
+_NOTE: on Windows, I used `mprof-report` from WSL/bash, since it appears to not be distributed in the Mono Windows installer._
+
+Your output will be something like:
+
+```
+Mono log profiler data
+	Profiler version: 3.0
+	Data version: 17
+	Arguments: log:calls,output=/data/user/0/com.xamarin.android.helloworld/files/.__override__/profile.mlpd
+	Architecture: armv8
+	Operating system: linux
+	Mean timer overhead: 47 nanoseconds
+	Program startup: Wed Sep  4 10:46:50 2019
+	Program ID: 11929
+	Server listening on: 44657
+
+JIT summary
+	Compiled methods: 1454
+	Generated code size: 459916
+	JIT helpers: 0
+	JIT helpers code size: 0
+
+GC summary
+	GC resizes: 0
+	Max heap size: 0
+	Object moves: 0
+
+Method call summary
+Total(ms) Self(ms)      Calls Method name
+     465        0          1 (wrapper runtime-invoke) <Module>:runtime_invoke_void_JnienvInitializeArgs* (object,intptr,intptr,intptr)
+     465        9          1 Android.Runtime.JNIEnv:Initialize (Android.Runtime.JnienvInitializeArgs*)
+     171        0          3 (wrapper native-to-managed) Android.Runtime.DynamicMethodNameCounter:3 (intptr,intptr,intptr)
+     171        0          3 (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter:3 (intptr,intptr,intptr)
+     170        2          3 Android.App.Activity:n_OnCreate_Landroid_os_Bundle_ (intptr,intptr,intptr)
+     162        6          1 Android.Runtime.AndroidRuntime:.ctor (intptr,intptr,bool,intptr,intptr)
+     153        2          3 HelloWorld.MainActivity:OnCreate (Android.OS.Bundle)
+     147        0         21 (wrapper runtime-invoke) object:runtime_invoke_void (object,intptr,intptr,intptr)
+     132        1         14 Java.Interop.JniPeerMembers:.ctor (string,System.Type)
+     124        0          8 (wrapper runtime-invoke) object:runtime_invoke_void (object,intptr,intptr,intptr)
+     117        7          1 Java.Interop.JniRuntime:.ctor (Java.Interop.JniRuntime/CreationOptions)
+     106        6         14 Java.Interop.JniPeerMembers:.ctor (string,System.Type,bool,bool)
+     105        0         57 Java.Interop.JniRuntime/JniTypeManager:GetTypeSignature (System.Type)
+     100        0          4 (wrapper runtime-invoke) <Module>:runtime_invoke_void_intptr_int_intptr_intptr_int (object,intptr,intptr,intptr)
+     100        2          4 Android.Runtime.JNIEnv:RegisterJniNatives (intptr,int,intptr,intptr,int)
+      99        1         57 System.Linq.Enumerable:FirstOrDefault<Java.Interop.JniTypeSignature> (System.Collections.Generic.IEnumerable`1<Java.Interop.JniTypeSignature>)
+      99        0         10 Java.Interop.JniEnvironment/InstanceMethods:CallNonvirtualVoidMethod (Java.Interop.JniObjectReference,Java.Interop.JniObjectReference,Java.Interop.JniMethodInfo,Java.Interop.JniArgumentValue*)
+      98        5         57 System.Linq.Enumerable:TryGetFirst<Java.Interop.JniTypeSignature> (System.Collections.Generic.IEnumerable`1<Java.Interop.JniTypeSignature>,bool&)
+```
```

Comment:
I think it might be useful to add a note here, that the times above include the time spent in JIT. And that this time to JIT might be around 50% of the measured time for methods which are called just once.